### ISSUE TITLE
https://registry-1.docker.io/v2/library/openjdk/manifests/8-jre-alpine: unauthorized: incorrect username or password

### ISSUE THREAD

**Issue by gmarquez87** [sitewhere/sitewhere#664]
Hi!,

I'm trying to running the Sitewhere project, but when I execute the gradle command to package the microservices into Docker images in MacOS

sudo ./gradlew clean dockerImage

I have the next error:

* What went wrong:
Execution failed for task ':service-asset-management:dockerImage'.
> Could not build image: Get https://registry-1.docker.io/v2/library/openjdk/manifests/8-jre-alpine: unauthorized: incorrect username or password

Reading other issues related to this problem, I found some possible solutions like: logout and login in your docker hub account (I tried this, but doesn't work), create another docker account (doesn't work), logout from using `docker logout` from terminal (doesn't work), logout from MacOS Docker app (doesn't work). At the end, I tried to logout, login, run gradle with/without login, but nothing works.

My current java version is:

```
gaston.marquez$ java -version
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
```

So, basically, I don't know what to do. I really appreciate your help on this issue.

Regards.

**Comment by derekadams** [comment]
From the error message, it looks like the base JRE image we use is not available on Docker Hub. We will try to replicate the issue.

**Comment by derekadams** [comment]
FYI, it sounds like @jorgevillaverde-sitewhere has been able to reproduce the issue on multiple operating systems. We are looking into it.

**Comment by derekadams** [comment]
You should be able to work around the issue by manually running `docker image pull openjdk:8-jre-alpine` to pull the base image first.

**Comment by derekadams** [comment]
It looks like the base image is only accessible via logged-in users, so you will need to add the following lines to your `~/.gradle/gradle.properties` file:

```
dockerRepository=docker.io
registryUsername=(your docker username)
registryPassword=(your docker password)
registryEmail=(your docker email address)
```

**Comment by gmarquez87** [comment]
Hi!, many thanks @derekadams  for your help. Is working now.

Finally, my last question. When I try to deploy the services in  ⁨sitewhere-recipes/docker-compose⁩/default⁩ I have the next error:

`default gaston.marquez$ docker-compose up
WARNING: The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Pulling instance-management (sitewhere/service-instance-management:2.0.0)...
ERROR: manifest for sitewhere/service-instance-management:2.0.0 not found`

The manifest is not available for some reason. I followed the instructions of SiteWhere Deployment Guide and before to deploy the services I deployed the infrastructure default, but still, the error appears.

Thanks.
